cmd2/decorators.py

```diff
@@ -17,12 +31,22 @@ def cat_decorator(func):
 
 
 def with_argument_list(*args: List[Callable], preserve_quotes: bool = False) -> Callable[[List], Optional[bool]]:
-    """A decorator to alter the arguments passed to a do_* cmd2 method. Default passes a string of whatever the user
-    typed. With this decorator, the decorated method will receive a list of arguments parsed from user input.
+    """
+    A decorator to alter the arguments passed to a ``do_*`` method. Default
+    passes a string of whatever the user typed. With this decorator, the
+    decorated method will receive a list of arguments parsed from user input.
 
-    :param args: Single-element positional argument list containing do_* method this decorator is wrapping
-    :param preserve_quotes: if True, then argument quotes will not be stripped
+    :param args: Single-element positional argument list containing ``do_*`` method
+                 this decorator is wrapping
+    :param preserve_quotes: if ``True``, then argument quotes will not be stripped
     :return: function that gets passed a list of argument strings
+
+    :Example:
```

Comment:
I really like the couple examples added to the decorators.  Consider adding examples to the other two decorators.